parcels/kernel.py

```diff
@@ -121,15 +122,6 @@ def _cache_key(self):
         key = self.name + self.ptype._cache_key + field_keys + ('TIME:%f' % ostime())
         return md5(key.encode('utf-8')).hexdigest()
 
-    @staticmethod
-    def fix_indentation(string):
-        """Fix indentation to allow in-lined kernel definitions."""
-        lines = string.split('\n')
-        indent = re.compile(r"^(\s+)").match(lines[0])
-        if indent:
-            lines = [line.replace(indent.groups()[0], '', 1) for line in lines]
-        return "\n".join(lines)
-
```

Comment:
Note that this would be a change to the public API. As mentioned https://github.com/OceanParcels/parcels/pull/1610#discussion_r1696604280 I don't see a usecase where a parcel user directly calls this code, so I don't think this removal would be disruptive.